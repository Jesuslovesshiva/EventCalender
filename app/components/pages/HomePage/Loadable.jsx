import PageLoadingIndicator from '@/components/atoms/PageLoadingIndicator'
import loadable from '@/utils/loadable'

export default loadable(() => import('./index'), {
  fallback: <PageLoadingIndicator />
})
