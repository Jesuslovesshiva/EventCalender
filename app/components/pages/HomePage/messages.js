import {defineMessages} from 'react-intl'

const scope = 'app'

export default defineMessages({
  someMessage: {
    id: `${scope}.someMessage`,
    defaultMessage: 'This is a message to show localization usage.',
  }
})
