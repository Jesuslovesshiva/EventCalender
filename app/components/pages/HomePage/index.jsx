import React, {useState} from 'react'
import {FormattedMessage} from 'react-intl'

import Button from '@/components/atoms/Button'
import SaveIcon from '@/components/atoms/Icon/Save'
import DateCyclePicker from '@/components/molecules/DateCyclePicker'
import Counter from '@/components/organisms/Counter'

import messages from './messages'

const DATE = new Date('2022-03-21')

const HomePage = () => {
  const [date, setDate] = useState(DATE)

  return (
    <div className="flex-1 flex flex-col justify-start p-6 bg-gradient-to-b bg-gradient-gray">
      <div className="mb-6">
        <h2 className="font-heading-large mb-1">
          Components you&apos;ll need:
        </h2>

        <div className="mb-1">
          <DateCyclePicker value={date} onChange={setDate} />
        </div>

        <div className="mb-1">
          <Button startIcon={<SaveIcon />}>
            Save
          </Button>
        </div>

        <div className="mb-1">
          <FormattedMessage {...messages.someMessage} />
        </div>
      </div>


      <div className="mb-6">
        <h2 className="font-heading-large mb-1">
          Random example component:
        </h2>

        <Counter />
      </div>
    </div>
  )
}

export default React.memo(HomePage)
