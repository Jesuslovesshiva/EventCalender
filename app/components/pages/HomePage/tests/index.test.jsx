import configureTestStore from '@/utils/test-utils/configure-test-store'
import render from '@/utils/test-utils/custom-render'

import HomePage from '../index'

jest.mock('@/components/organisms/Counter', () => () => <div>Counter</div>)

const {clearDispatch, options} = configureTestStore()
const renderComponent = () => render(<HomePage />, options)

describe('HomePage', () => {
  beforeEach(clearDispatch)

  it('should match the snapshot', () => {
    const {container} = renderComponent()
    expect(container.firstChild).toMatchSnapshot()
  })
})
