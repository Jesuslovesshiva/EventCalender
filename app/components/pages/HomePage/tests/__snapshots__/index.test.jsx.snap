// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage should match the snapshot 1`] = `
<div
  class="flex-1 flex flex-col justify-start p-6 bg-gradient-to-b bg-gradient-gray"
>
  <div
    class="mb-6"
  >
    <h2
      class="font-heading-large mb-1"
    >
      Components you'll need:
    </h2>
    <div
      class="mb-1"
    >
      <div>
        <div
          class="border border-gray-medium divide-x divide-gray-medium rounded inline-flex transition cursor-pointer hover:border-white hover:divide-white hover:bg-silver hover:shadow-md specific:divider-gray-light"
        >
          <button
            aria-label="vorherige Woche"
            class="w-[26px] transition-[border-color,color] text-gray-medium hover:text-gray-dark"
          >
            <div
              class="fill-current w-[1em] h-[1em] inline-block text-lg"
              data-icon="arrow_back_ios"
            />
          </button>
          <div
            class="flex items-center py-1 transition pl-1.5 pr-1.5"
          >
            <div
              class="fill-current w-[1em] h-[1em] inline-block text-lg mr-1"
              data-icon="date_range"
            />
            <input
              aria-label="von"
              class=" w-[75px] rounded-sm transition-[outline-color,background-color] bg-transparent font-bold text-center outline-transparent focus-visible:bg-silver focus-visible:outline focus-visible:outline-1 focus-visible:outline-gray-medium focus-visible:outline-offset-0 "
              placeholder="TT.MM.JJJJ"
              value="21.03.2022"
            />
            <div
              class="w-3.5 flex justify-center items-center font-bold"
            >
              -
            </div>
            <input
              aria-label="bis"
              class=" w-[75px] rounded-sm transition-[outline-color,background-color] bg-transparent font-bold text-center outline-transparent focus-visible:bg-silver focus-visible:outline focus-visible:outline-1 focus-visible:outline-gray-medium focus-visible:outline-offset-0 "
              placeholder="TT.MM.JJJJ"
              value="27.03.2022"
            />
            <button
              aria-label="Kalendar ein-/ausschalten"
              class="transition text-gray-medium hover:text-gray-dark"
            >
              <div
                class="fill-current w-[1em] h-[1em] inline-block text-2xl"
                data-icon="arrow_drop_down"
              />
            </button>
          </div>
          <button
            aria-label="nÃ¤chste Woche"
            class="w-[26px] transition-[border-color,color] text-gray-medium hover:text-gray-dark"
          >
            <div
              class="fill-current w-[1em] h-[1em] inline-block text-lg rotate-180"
              data-icon="arrow_back_ios"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="mb-1"
    >
      <button
        class="inline-flex justify-center gap-1 font-button-text items-center transition focus-visible:outline-dashed focus-visible:outline-primary focus-visible:outline-offset-1 disabled:pointer-events-none specificity-up:bg-blue-100 min-h-[28px] min-w-[28px] text-white relative overflow-hidden isolate before:bg-gray-dark before:absolute before:inset-0 before:z-[-1] before:opacity-0 before:transition-opacity hover:shadow-md hover:before:opacity-100 disabled:bg-gray-medium disabled:bg-none disabled:text-gray-light disabled:shadow-none shadow bg-gradient-to-l bg-gradient-primary rounded px-3 py-[0.4375rem] "
        type="button"
      >
        <div
          class="flex text-lg"
        >
          <div
            class="fill-current w-[1em] h-[1em] inline-block"
            data-icon="save"
          />
        </div>
        Save
      </button>
    </div>
    <div
      class="mb-1"
    >
      This is a message to show localization usage.
    </div>
  </div>
  <div
    class="mb-6"
  >
    <h2
      class="font-heading-large mb-1"
    >
      Random example component:
    </h2>
    <div>
      Counter
    </div>
  </div>
</div>
`;
